### ISSUE TITLE
Story / Make hight of content hug the content

### ISSUE THREAD

**Issue by johnsBeharry** [peakshift/makers.bolt.fun#220]
Current behaviour

<img width="1305" alt="Screenshot 2023-01-10 at 17 37 15" src="https://user-images.githubusercontent.com/183140/211609545-e952e4d9-c3b1-4a71-9bdc-1ba39e6ac5d8.png">

_i think this layout behaviour is because the columns are set using grid instead of flex columns._

Expected behaviour
- there is only the bottom padding between the last line of the story content and the border.

**Comment by MTG2000** [comment]
Hmmmm
Yeah, you are right, this behaviour is due to using CSS grids.
But the reason why I'm using Grid not Flex is because the order of elements is not fixed across different screen sizes.
Flexbox assumes that elements will stay in the same order (vertically OR horizontally), but in our case, elements from the side bars will be mixed with the middle column on mobile.

Layout order:
on mobile:
```
"main content"
"actions (vote)"
"author card"
"comments"
"trending"
```

on medium screen:
```
"actions content"
"............... author"
"............... comments"
```

on desktop:
```
"actions content author"
".............. comments ........"
".............. ..................... ........."
```

So that's why Grid is needed.

If this problem where the story is very short is really serious, then there is 2 possible solutions we can do:
1- Make the story content card take its actual width & leave the rest of the row as empty space like this:
![image](https://user-images.githubusercontent.com/50504183/211754714-346d8dd8-5792-4ebc-b849-333088943d8f.png)

2- The other option would be to duplicate the code.
Basically I just create 2 different components, one for the desktop screen & the other for mobile screen.
This would make it appear as we want it to, but it would produce a lot of duplication in the code, because the 2 components would largely be identical, just different ordering.

So what do you suggest?
1, 2, or leave-as-is??

**Comment by johnsBeharry** [comment]
From what I recall flex box does not assume things need to stay in the same order. I'm almost certain this can be controlled by a flex box property.

From the tailwind docs: https://tailwindcss.com/docs/order

**Comment by MTG2000** [comment]
> From the tailwind docs: https://tailwindcss.com/docs/order

This is a little bit different...

The flexbox ordering is item-based.
Meaning that you can give a number to specify which items appear before/after other items.
But in our case, the ordering is not just items based, it's first column based, THEN item based.
We need to say that:
we have 3 columns. column 1 has items: x, y, z in this order. column 2 has items: q, s, ..etc. column3 has items: ...

So this is something can't be done using Flexbox. It's something for Grid.

There is another option in CSS which is: CSS columns.
This allows us to sort COLUMN-BASED , but no items based, so I don't think it can help us.

**Comment by barefoot-88** [comment]
Would this issue be solved if we sacrificed (hide) a couple of components for the mobile version?

**Back button**
This isn’t needed on mobile as web browsers often have them inbuilt. Same with Android devices.

**Vote button**
Users can’t vote on mobile due to lack of WebLN. They could use a WebLN enabled browser, but the UX of these is pretty crap anyway so don’t think it’s a use case we should prioritise.

**Author card**
Not essential for mobile viewing. But since it’s last in the horizontal grid we could keep it?

**Comment by MTG2000** [comment]
Yeah, maybe...

But is this really the best solution??
Give me some time & I'll look to see if there is a better solution without much sacrificing.

**Comment by MTG2000** [comment]
I've pushed the fix in PR #223
